cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -390,6 +392,61 @@ This ensures that the next time the MapReduce job runs, it processes only the ne
     partitionCommitter.onFinish(succeeded);
   }
 
+Processing using a Generic
+--------------------------
+Partitions of a partitioned file set can also be incrementally processed from other program types
+using the generic ``PartitionConsumer`` APIs. The implementation of these APIs that can be used from multiple instances
+of a program is ``ConcurrentPartitionConsumer``. To use, you simply need to provide the instance of the
+partitioned file set you want to consume from, along with a ``StatePersistor``, responsible for managing
+persistence of the consumer's state::
+
+  // This can be in any program where we have access to Datasets,
+  // such as a Worker, Workflow Action, or even in a MapReduce
+  PartitionConsumer consumer =
+    new ConcurrentPartitionConsumer(partitionedFileSet, new CustomStatePersistor(persistenceTable));
+
+  // Call consumePartitions to get a list of partitions to process
+  final List<PartitionDetail> partitions = partitionConsumer.consumePartitions().getPartitions();
+
+  // Process partitions
+
```

Comment:
Yea, it was meant to indicate any custom processing by the user of these partitions.